tests/Backend/Snowflake/TypedTableInWorkspaceTest.php

```diff
@@ -109,7 +109,249 @@ public function testUnloadFromWSToTypedTable(): void
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => $tableId,
+            'deduplicationStrategy' => DeduplicationStrategy::INSERT->value,
+        ]);
+        $eventAssertCallback = function ($events) {
+            $this->assertCount(1, $events);
+            $this->assertArrayHasKey('performance', $events[0]);
+            $this->assertIsArray($events[0]['performance']);
+            $this->assertArrayHasKey('importDecomposed', $events[0]['performance']);
+            $this->assertCount(1, $events[0]['performance']['importDecomposed']);
+            $this->assertArrayHasKey('name', $events[0]['performance']['importDecomposed'][0]);
+            // there is only one stage in full load
+            $this->assertSame('ctasLoad', $events[0]['performance']['importDecomposed'][0]['name']);
+
+            // set event as last event
+            $this->lastEventId = $events[0]['uuid'];
+        };
+        $query = new EventsQueryBuilder();
+        $query->setEvent('storage.tableImportDone')
+            ->setTokenId($this->tokenId)
+            ->setRunId($runId);
+        $this->assertEventWithRetries($this->_client, $eventAssertCallback, $query);
+
+        $expectedFullLoad = [
+            [
+                [
+                    'columnName' => 'id',
+                    'value' => '1',
+                    'isTruncated' => false,
+                ],
+                [
+                    'columnName' => 'name',
+                    'value' => 'john',
+                    'isTruncated' => false,
+                ],
+            ],
+            [
+                [
+                    'columnName' => 'id',
+                    'value' => '2',
+                    'isTruncated' => false,
+                ],
+                [
+                    'columnName' => 'name',
+                    'value' => 'does',
+                    'isTruncated' => false,
+                ],
+            ],
+            [
+                [
+                    'columnName' => 'id',
+                    'value' => '2',
+                    'isTruncated' => false,
+                ],
+                [
+                    'columnName' => 'name',
+                    'value' => 'doesToo',
+                    'isTruncated' => false,
+                ],
+            ],
+        ];
+
+        /** @var array $data */
+        $data = $this->_client->getTableDataPreview($this->tableId, [
+            'format' => 'json',
+        ]);
+        self::assertEquals($expectedFullLoad, $data['rows']);
+
+        // do incremental load
+        // incremental load also does not deduplicate values
+        // this will cause that all values are present twice
+        $this->_client->writeTableAsyncDirect($this->tableId, [
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => $tableId,
+            'incremental' => true,
+            'deduplicationStrategy' => DeduplicationStrategy::INSERT->value,
```

Comment:
Taky by to mělo valit takto.
```suggestion
            'deduplicationStrategy' => DeduplicationStrategy::INSERT,
```